tests/sentry/integrations/slack/test_unfurl.py

```diff
@@ -0,0 +1,97 @@
+import pytest
+from django.test import RequestFactory
+
+from sentry.incidents.logic import CRITICAL_TRIGGER_LABEL
+from sentry.integrations.slack.message_builder.incidents import build_incident_attachment
+from sentry.integrations.slack.message_builder.issues import build_group_attachment
+from sentry.integrations.slack.unfurl import LinkType, UnfurlableUrl, link_handlers, match_link
+from sentry.models import Integration, OrganizationIntegration
+from sentry.testutils import TestCase
+from sentry.testutils.helpers.datetime import before_now, iso_format
+from sentry.utils.compat import filter
+from sentry.utils.compat.mock import Mock, patch
+
+
+@pytest.mark.parametrize(
+    "url,expected",
+    [
+        ("http://invalid_link", (None, None)),
+        (
+            "https://sentry.io/organizations/org1/issues/12345/",
+            (LinkType.ISSUES, {"issue_id": 12345, "event_id": None}),
+        ),
+        (
+            "https://sentry.io/organizations/org1/alerts/rules/details/12345/",
+            (LinkType.INCIDENTS, {"incident_id": 12345, "org_slug": "org1"}),
+        ),
+    ],
+)
+def test_match_link(url, expected):
+    assert match_link(url) == expected
+
+
+class UnfurlTest(TestCase):
```

Comment:
This covers the testing removed from the `LinkSharedEventTest` for issue and incident unfurling.